### PR TITLE
fix(fsc): Fixes to test is_feature_enabled with fsc.

### PR DESCRIPTION
**Pull request by yasirfolio3**

### CODE REVIEW

File: optimizely/utils/types.go [hunk 1]
```diff
@@ -26,41 +26,49 @@ var intType = reflect.TypeOf(int64(0))
...
 
 // GetIntValue will attempt to convert the given value to an int64
 func GetIntValue(value interface{}) (int64, error) {
-	v := reflect.ValueOf(value)
-	if v.Type().String() == "int64" || v.Type().ConvertibleTo(intType) {
-		intValue := v.Convert(intType).Int()
-		return intValue, nil
+	if value != nil {
+		v := reflect.ValueOf(value)
+		if v.Type().String() == "int64" || v.Type().ConvertibleTo(intType) {
```
[the remark this answers is not in the document]
Comment (pawels-optimizely) [reply optimizely/utils/types.go]: ok, so we can leave it like this, or convert it to float and then cast it to int specifically.

File: optimizely/config/datafileProjectConfig/mappers/condition_trees.go [hunk 2]
```diff
@@ -28,7 +28,16 @@ var ErrEmptyTree = errors.New("Empty Tree")
 
 // Takes the conditions array from the audience in the datafile and turns it into a condition tree
 func buildConditionTree(conditions interface{}) (conditionTree *entities.TreeNode, retErr error) {
-	value := reflect.ValueOf(conditions)
+
+	var parsedConditions interface{}
```
Comment (mikeng13) [inline optimizely/config/datafileProjectConfig/mappers/condition_trees.go]: Can you add unit tests for this please?

File: optimizely/decision/evaluator/condition.go [hunk 3]
```diff
@@ -47,6 +48,9 @@ func (c CustomAttributeConditionEvaluator) Evaluate(condition entities.Condition
 
 	var matcher matchers.Matcher
 	matchType := condition.Match
+	if matchType == "" {
```
Comment (mikeng13) [inline optimizely/decision/evaluator/condition.go]: Please add a unit test for this

File: optimizely/utils/types.go [hunk 4]
```diff
@@ -26,41 +26,49 @@ var intType = reflect.TypeOf(int64(0))
 
 // GetBoolValue will attempt to convert the given value to a bool
 func GetBoolValue(value interface{}) (bool, error) {
-	v := reflect.ValueOf(value)
-	if v.Type().String() == "bool" {
-		return v.Bool(), nil
+	if value != nil {
```
Comment (mikeng13) [inline optimizely/utils/types.go]: Please update the unit tests to account for these scenarios.

Comment (yasirfolio3) [reply optimizely/utils/types.go]: Test-cases implemented, please review.

File: optimizely/config/datafileProjectConfig/mappers/condition_trees_test.go [hunk 5]
```diff
@@ -60,6 +61,45 @@ func TestBuildAudienceConditionTreeSimpleAudienceCondition(t *testing.T) {
 	assert.Equal(t, expectedConditionTree, conditionTree)
 }
 
+func TestBuildConditionTreeUsingDatafileAudienceConditions(t *testing.T) {
+	datafile, err := resources.GetTestDataFileJSON("bucketing_id.json")
```
Comment (mikeng13) [inline optimizely/config/datafileProjectConfig/mappers/condition_trees_test.go]: We shouldn't need to load a full datafile for this. Instead just construct the `datafile.Audiences` array as a templated string like: 

```
datafileAudience = datafile.Audience{
  Conditions...
}
```

File: optimizely/resources/testdatafiles/bucketing_id.json [hunk 6]
```diff
@@ -0,0 +1,355 @@
+{
```
Comment (mikeng13) [inline optimizely/resources/testdatafiles/bucketing_id.json]: Let's not introduce a full datafile yet. I'd like to keep the code simpler for now.

File: optimizely/utils/types_test.go [hunk 7]
```diff
@@ -0,0 +1,235 @@
+/****************************************************************************
```
Comment (mikeng13) [inline optimizely/utils/types_test.go]: Thanks for adding these tests

File: optimizely/config/datafileProjectConfig/mappers/condition_trees_test.go (hunk 5 above)
Comment (yasirfolio3) [reply optimizely/config/datafileProjectConfig/mappers/condition_trees_test.go]: Recommended changes made.

File: optimizely/resources/helper_methods.go [hunk 8]
```diff
@@ -0,0 +1,45 @@
+/****************************************************************************
```
Comment (mikeng13) [inline optimizely/resources/helper_methods.go]: Please remove this file,.

Comment (yasirfolio3) [reply optimizely/resources/helper_methods.go]: done.

**Comment by mikeng13** [comment]
@yasirfolio3 unit tests are failing. Can you look into why?

**Comment by yasirfolio3** [comment]
> @yasirfolio3 unit tests are failing. Can you look into why?

done, because now we are providing support for pre v3.0 audiences as well, one of the tests had audience mapping increased from 3 to 4.

**Pull request merged**